### PR TITLE
Add manual upgrade steps

### PR DESCRIPTION
**Pull request by bk201**
**Related issue**
https://github.com/harvester/harvester/issues/1665

### CODE REVIEW

File: docs/upgrade.md [hunk 1]
```diff
@@ -14,5 +14,311 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
...
+    - <span style="color:#DE3163">**On the controller node**</span>, drain pods out of node:
+
+        ```bash
+        # ** On the controller node **
+        sudo -i kubectl drain <node> --force --ignore-daemonsets --delete-local-data
+        ```
+
+        Wait until pods are evicted.
+
+    - ** On the upgrading node **, pre-load images:
```
Comment (weihanglo) [inline docs/upgrade.md:143]:
```suggestion
    - **On the upgrading node**, pre-load images:
```

File: docs/upgrade.md [hunk 2]
```diff
@@ -14,5 +14,311 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
...
+
+        ```bash
+        sudo -i /usr/local/harvester-upgrade/upgrade-helpers/bin/harv-load-images /usr/local/harvester-upgrade/iso_mnt
+        ```
+
+    - ** On the upgrading node **, remove old manifests and charts:
```
Comment (weihanglo) [inline docs/upgrade.md:149]:
```suggestion
    - **On the upgrading node**, remove old manifests and charts:
```

File: docs/upgrade.md [hunk 3]
```diff
@@ -14,5 +14,311 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
...
+        sudo rm -f /var/lib/rancher/rke2/server/static/charts/rancher-monitoring-*.tgz
+        ```
+
+        Copy new charts:
+
+        ```bash
+        sudo cp /usr/local/harvester-upgrade/iso_mnt/bundle/harvester/static/charts/* /var/lib/rancher/rke2/server/static/charts/
+        ```
+
+    - Upgrade OS and reboot.  ** On the upgrading node **, run:
```
Comment (weihanglo) [inline docs/upgrade.md:167]:
```suggestion
    - Upgrade OS and reboot.  **On the upgrading node**, run:
```

File: docs/upgrade.md [hunk 4]
```diff
@@ -14,5 +14,311 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
...
+
+        ```bash
+        sudo mkdir -p /usr/local/harvester-upgrade/squashfs_mnt
+        sudo mount /usr/local/harvester-upgrade/iso_mnt/rootfs.squashfs /usr/local/harvester-upgrade/squashfs_mnt
+        sudo /usr/local/harvester-upgrade/upgrade-helpers/bin/harv-os-upgrade --directory /usr/local/harvester-upgrade/squashfs_mnt
+        sudo reboot
+        ```
+
+    - After the upgrading node is rebooted, verify versions.  ** On the upgrading node **, run:
```
Comment (weihanglo) [inline docs/upgrade.md:176]:
```suggestion
    - After the upgrading node is rebooted, verify versions.  **On the upgrading node**, run:
```

**Review by weihanglo — COMMENTED** [review]
I felt somehow astray when navigating the guide. Are those steps need to be perform in sequence? If yes I think ordered lists would be more suitable than unordered bullets.

File: docs/upgrade.md [hunk 5]
```diff
@@ -14,5 +14,311 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
...
+node4   Ready    <none>                      112m   v1.21.5+rke2r1
+```
+
+Download the helpers to the controller node:
+
+```bash
+sudo mkdir -p /usr/local/harvester-upgrade
+sudo chown rancher:rancher /usr/local/harvester-upgrade
+cd /usr/local/harvester-upgrade
+curl -fL https://github.com/bk201/upgrade-helpers/releases/latest/download/upgrade-helpers.tar.gz | tar xzvf -
```
[the remark this answers is not in the document]
Comment (bk201) [reply docs/upgrade.md:64]: > Also, it'd be better to put it under harvester organization than personal repo.


That will be migrated if the doc is ready.

[the remark this answers is not in the document]
Comment (bk201) [reply docs/upgrade.md]: Fixed.

File: docs/upgrade.md (hunk 1 above)
Comment (bk201) [reply docs/upgrade.md]: Thanks, fixed.

**Review by nunix — COMMENTED** [review]
Rephrased the descriptions and standardized the `Example output`

File: docs/upgrade.md [hunk 6]
```diff
@@ -14,5 +14,201 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
...
+
+
+## Manual Upgrade
+
+!!!warning
+    The steps to perform a manual upgrade is only recommended for experienced users. Use it at your own risk.
+
+### Overview
+
+The manual upgrade process consists these steps:
```
[the remark this answers is not in the document]
Comment (btat) [reply docs/upgrade.md:27]: Should be "consists of".

[the remark this answers is not in the document]
Comment (nunix) [reply docs/upgrade.md:27]: thanks Billy, I corrected the suggestion

File: docs/upgrade.md [hunk 7]
```diff
@@ -14,5 +14,201 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
...
+- Upgrade the RKE2 runtime.
+- Upgrade the Harvester service.
+
+### Preparation
+
+- Backup your VMs.
+
+#### Get the new release
+
+Download the ISO of a new release from Harvester release page and verify the checksum of ISO first.
```
[the remark this answers is not in the document]
Comment (jtravee) [reply docs/upgrade.md:41]: Minor nit: There is an extra space in between "the" and "[".

**Review by jtravee — CHANGES_REQUESTED** [review]

File: docs/upgrade.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (nunix) [reply docs/upgrade.md:41]: thank you Jen :) ... updated

**Review by nunix — COMMENTED** [review]
removed extra space as per Jen's feedback

**Comment by bk201** [comment]
Thanks, @nunix, @jtravee, and @btat I commit the suggested changes.

**Review by nunix — APPROVED** [review]

File: docs/upgrade.md [hunk 8]
```diff
@@ -14,5 +14,202 @@ Description: Harvester provides two ways to upgrade. Users can either upgrade us
 !!!note
     Upgrade is not supported from previous versions to v0.3.0 version.
 
-    - A manual upgrade process starting with v0.3.0 is being investigated. Harvester will inform the community once this process is in place.
     - One-click upgrade will be supported starting with the v1.0.0 release.
+
+
+## Manual Upgrade
```
Comment (guangbochen) [inline docs/upgrade.md:20]: should we specify if the Manual Upgrade only supports from `v0.3.0` to `v1.0.0`?
e.g., # Manual Upgrade(From v0.3.0 to v1.0.0)

Comment (bk201) [reply docs/upgrade.md]: Fixed in 
https://github.com/harvester/docs/pull/67/commits/85589ed563a4486a27409f853207f2af355848b3#diff-9cb3d19a7e30c718742ca50bf7a18ed9a3d057d82eb76fa410358b5b60900e12

**Comment by guangbochen** [comment]
thanks, and verified both manual and automatic upgrade doc path works.
image

**Pull request merged**